src/screens/AddNewTerm.js

```diff
@@ -0,0 +1,253 @@
+    LANG_DATA[ENGLISH_PHRASE_HEADING][nativeLanguage];
+  const malagasyPhraseHeading =
+    LANG_DATA[MALAGASY_PHRASE_HEADING][nativeLanguage];
+  const placeholderNewTerm = LANG_DATA[PLACEHOLDER_NEW_TERM][nativeLanguage];
+  const addButtonText = LANG_DATA[ADD_BUTTON_TEXT][nativeLanguage];
+  const selectCategoryHeading =
+    LANG_DATA[SELECT_CATEGORY_HEADING][nativeLanguage];
+  const closeButtonText = LANG_DATA[CLOSE_BUTTON_TEXT][nativeLanguage];
+
+  const selectCategory = category => {
+    setSelectedCategory(category);
+    setCategoryId(category?.id);
+    setOpenCategoriesList(false);
+  };
+
+  const addNewTermToCategory = () => {
+    const newTerm = {
+      catId: categoryId,
+      id: generateId(),
+      name: {
+        en: englishPhrase,
+        mg: malagasyPhrase,
+      },
+    };
+    addNewTerm(newTerm);
+    setEnglishPhrase('');
+    setMalagasyPhrase('');
+    setSelectedCategory({});
+    setCategoryId('');
+  };
+
+  return (
+    <SafeAreaView style={{flex: 1}}>
+      <KeyboardAvoidingView style={{flex: 1}}>
+        <View style={{paddingHorizontal: 35, paddingVertical: 23}}>
+          <View style={styles.header}>
+            <ToolBar
+              button={
+                <ToolButton
+                  onPress={() => {
+                    navigation.navigate('Home');
+                  }}>
+                  <BackIcon width={24} height={24} fill="#FFFFFF" />
+                </ToolButton>
+              }
+            />
+            <ToolBar
+              button={
+                <LanguageSwitcher
+                  firstLanguage={LANGUAGE_NAMES.EN}
+                  LeftText={nativeLanguage === LANGUAGE_NAMES.EN ? 'MG' : 'EN'}
+                  RightText={nativeLanguage === LANGUAGE_NAMES.EN ? 'EN' : 'MG'}
+                  color="#FFFFFF"
+                  iconType=""
+                  iconName="swap-horiz"
+                  onPress={() =>
+                    setLanguageName(
+                      nativeLanguage === LANGUAGE_NAMES.EN
+                        ? LANGUAGE_NAMES.MG
+                        : LANGUAGE_NAMES.EN,
+                    )
+                  }
+                  iconSize={24}
+                />
+              }
+            />
+            <ToolBar
+              button={
+                <ToolButton onPress={() => {}}>
+                  <ModeIcon width={24} height={24} fill="#FFFFFF" />
+                </ToolButton>
+              }
+            />
+          </View>
+          <View style={styles.heading}>
+            <SectionHeading text={categoryHeading} />
+            <TouchableOpacity
+              style={styles.select}
+              onPress={() => setOpenCategoriesList(true)}>
+              <Text
+                numberOfLines={1}
+                ellipsizeMode={'tail'}
+                style={styles.labelSelect}>
+                {Object.keys(selectedCategory).length === 0 &&
+                selectedCategory.constructor === Object
```

Comment:
its easier to do this check `!selectedCategory?.name ? .............`

Reply:
Thanks!